xwiki-platform-core/xwiki-platform-index/xwiki-platform-index-ui/src/main/resources/XWiki/DeletedDocuments.xml

```diff
@@ -341,6 +342,48 @@ XWiki.index.trash.documents.confirmDelete = function(event, table, rowIdx) {
       );
   }
 }
+
+/**
+ * Event listener, called when clicking the "delete all" button.
+ *
+ * @param event the link activation event
+ */
+XWiki.index.trash.documents.confirmDeleteAll = function(event) {
+  event.stop();
+  if (this.hasClassName('disabled')) {
+    // Do nothing if the button was already clicked and it's waiting for a response from the server.
+    return;
+  } else {
+    new XWiki.widgets.ConfirmedAjaxRequest(
+      /* Ajax request URL */
+      this.href = XWiki.currentDocument.getURL('delete'),
+      /* Ajax request parameters */
+      {
+        parameters : {
+          'confirm': 1,
+          'emptybin': 'true',
```

Comment:
Any reason to use a string rather than directly the boolean value?

Reply:
None. I can use a boolean.